MotionPhotoReader/app/src/main/java/com/example/motionphotoreader/MotionPhotoInfo.java

```diff
@@ -0,0 +1,109 @@
+package com.example.motionphotoreader;
+
+import android.media.MediaExtractor;
+import android.media.MediaFormat;
+
+import com.adobe.internal.xmp.XMPException;
+import com.adobe.internal.xmp.XMPMeta;
+
+import java.io.File;
+import java.io.FileDescriptor;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * Holds data about the Motion Photo file.
+ */
+public class MotionPhotoInfo {
+
+    private final int width;
+    private final int height;
+    private final long duration;
+
+    private final int videoOffset;
+    private final long presentationTimestampUs;
+
+    private MediaExtractor extractor;
```

Comment:
make this `final`, it should only be set in the constructor

Reply:
(Same question as above).